### PR TITLE
jetpack: Replace unnecessary `page` dep

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

We haven't been using this library correctly in the first place, and it
has now broken the incorrect way we were using it (see #21117).

It seems that the way we were using it works out as equivalent to a
single line of JS, so just do that.

#### Jetpack product discussion

p1631824628152000/1544445965.249000-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Is CI happy?
* Does the relevant stuff still work?

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/mixins/url-search/index.js [hunk 1]
```diff
@@ -47,10 +46,10 @@ export default {
 		debug( 'search posts for:', keywords );
 		if ( this.props.search && keywords ) {
 			debug( 'replacing URL: ' + searchURL );
-			page.replace( searchURL );
+			window.location.href = searchURL;
 		} else {
 			debug( 'setting URL: ' + searchURL );
-			page( searchURL );
+			window.location.href = searchURL;
```
Comment (zinigor) [inline projects/plugins/jetpack/_inc/client/mixins/url-search/index.js:52]: Whoa, this is definitely better if it's working the same way we expect it to. Unless I'm missing something - I don't remember why we were using `Page` here anyway.
A question though - do we need this `if` still? The only thing I see different is the debug statement, which we can probably make the same too.

Comment (anomiex) [reply projects/plugins/jetpack/_inc/client/mixins/url-search/index.js:52]: I have no idea whether the different debug message holds any useful information for anyone. It seems to differentiate between doing a search and clearing a search?

As for the origin, it looks like it got copied into Jetpack in #8208 from automattic/dops-components. That seems to have copied it from Calypso in Automattic/dops-components#93. Calypso actually does use page.js for routing, according to various mentions in its README files (e.g. [here](https://github.com/Automattic/wp-calypso/blob/1276e7ecb60e62b3784098c411718bafae2e6b7c/docs/routing.md#route-definitions)).

**Review by jeherve — APPROVED** [review]
This tests well for me. ✅

**Comment by anomiex** [comment]
r232053-wpcom

**Pull request merged**